building/glass/branches-schema.yaml

```diff
@@ -2,42 +2,37 @@ $schema: "http://json-schema.org/draft-07/schema#"
 title: BranchesConfig
 type: object
 properties:
-  upload-targets:
-    type: array
-    items:
-      type: object
-      oneOf:
-        - properties:
-            name:
-              type: string
-            method:
-              enum: ["google-cloud-storage"]
-          required: ["name", "method"]
-          additionalProperties: false
-        - properties:
-            name:
-              type: string
-            method:
-              enum: ["rsync"]
-            user:
-              type: string
-            dir:
-              type: string
-          required: ["name", "method", "user", "dir"]
-          additionalProperties: false
   branches:
     type: array
     items:
       type: object
       properties:
         name:
           type: string
-          pattern: "^[0-9a-zA-Z_.-][0-9a-zA-Z_./-]+[0-9a-zA-Z_.-]$"
+          pattern: "[0-9a-zA-Z_-]+"
+        download:
+          type: string
+          pattern: "^no_download$|^https?://[0-9a-zA-Z_.-]+(:[0-9]+)?(/[0-9a-zA-Z_./-]*[0-9a-zA-Z_.-])?$"
```

Comment:
It might be better to just leave the download field blank if no download is intended, and fill in the fake value in Python. Otherwise, this field should be marked required.

Reply:
marked required since "no download" is an exceptional case for CI only